### PR TITLE
sys/random/minstd: fix signedness

### PR DESCRIPTION
**Pull request by PeterKietzmann**
### Contribution description

The Park-Miller LCG has a period of 2<sup>31</sup>-1. In RIOT, we call the generator twice to get one 32bit integer, thus, at least half as many numbers should be generated before duplicates occur (2<sup>30</sup>-1). However, the signedness of [`test`](https://github.com/RIOT-OS/RIOT/blob/8136edca9572c6414450ed9e3fa1eff4c5919754/sys/random/minstd.c#L46) prevents negative values and as a consequence, the state [is never incremented](https://github.com/RIOT-OS/RIOT/blob/master/sys/random/minstd.c#L52) which leads to a heavily reduced period. This PR changes `test` to a signed variable to fix that.

See [wikipedia](https://en.wikipedia.org/wiki/Lehmer_random_number_generator) for reference.

### Testing procedure
- `USEMODULE=prng_minstd make -C tests/rng/ clean all term`
- `dump 500000` and pipe the output for later analysis.
**Currently**
- With `seed=0` (default), the sequence repeats with a period of 223717, after the first 102819 dumped integers.
**With this PR**
- Outputs should repeat only after 2<sup>30</sup>-1 requests.

### CODE REVIEW

**Comment by tcschmidt** [comment]
@PeterKietzmann I would suggest to remove the following uninformed statement from the comments:

>  * While not very random when considering crypto requirements, this is probably
>  * random enough anywhere where pseudo-randomness is sufficient, e.g., when
>  * provided with a sensible seed source, for MAC algorithms.

**Comment by PeterKietzmann** [comment]
Fine with me. I've added a commit.

File: sys/random/minstd.c [hunk 1]
```diff
@@ -43,7 +38,7 @@ int rand_minstd(void)
 {
```
Comment (kaspar030) [inline sys/random/minstd.c:38]: The return value should be `int32_t` as well.

Comment (PeterKietzmann) [reply sys/random/minstd.c:38]: Sure... I amended the change to the first commit.

**Comment by PeterKietzmann** [comment]
@kaspar030 are you fine with this PR now?

**Pull request merged**